### ISSUE TITLE
hmac_totp() crashes in EVP_Q_mac() on Sparc platform with OpenSSL 3.0.x

### ISSUE THREAD

**Issue by tio-checo** [cisco/libacvp#853]
When compiling and running acvp_app on Sparc platform with OpenSSL 3.0.x, the application crashed in hmac_totp() function when calling EVP_Q_mac(). I narrowed down the issue to 'int len' variable. It is defined as int (4 bytes long), but pointer to long unsigned int (8 bytes long) is passed to EVP_Q_mac(). That triggers crash on Sparc when trying to write 8 bytes to 4 byte variable. Following patch fixed the problem:

```
diff --git a/libacvp-2.1.0/app/app_utils.c b/libacvp-2.1.0/app/app_utils.c
--- a/libacvp-2.1.0/app/app_utils.c
+++ b/libacvp-2.1.0/app/app_utils.c
@@ -162,12 +162,12 @@
                      int hash_max,
                      const char *md_name,
                      unsigned int key_len) {
-    int len = 0;
+    size_t len = 0;
     unsigned char buff[MAX_LEN];

-    EVP_Q_mac(NULL, "HMAC", NULL, md_name, NULL, key, key_len, msg, T_LEN, buff, MAX_LEN, (long unsigned int *)&len);
+    EVP_Q_mac(NULL, "HMAC", NULL, md_name, NULL, key, key_len, msg, T_LEN, buff, MAX_LEN, &len);
     memcpy_s(hash, hash_max, buff, len);
-    return len;
+    return (int)len;
 }
 #endif
```

**Comment by abkarcher** [comment]
That makes sense;  I will get this resolved ASAP. Thanks for reporting!

**Comment by tio-checo** [comment]
Sure, glad to help!

**Comment by abkarcher** [comment]
Addressed in #854, will be in an upcoming release. Thanks again!

**Comment by tio-checo** [comment]
You are welcome. Thank you for addressing it so quickly!
